lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -3284,6 +3301,34 @@ static void printElidePortAnnotations(OpAsmPrinter &p, Operation *op,
   printElideAnnotations(p, op, attr, elidedAttrs);
 }
 
+//===----------------------------------------------------------------------===//
+// NameKind Custom Directive
+//===----------------------------------------------------------------------===//
+
+static ParseResult parseNameKind(OpAsmParser &parser,
+                                 firrtl::NameKindEnumAttr &result) {
+  StringRef keyword;
+
+  if (!parser.parseOptionalKeyword(&keyword,
```

Comment:
do any of the parsers have the possibility of being confused by something whose name is "interesting_name"?

Reply:
No, I don't think so. A mnemonic or symbol name can be a precedence of nameKind, but "interesting_name" cannot be mnemonic or symbols.